qgis-samgeo-plugin/_samgeo_lib.py

```
@@ -0,0 +1,176 @@
+    orig_sys_path = list(sys.path)
+    orig_samgeo_mod = sys.modules.get("samgeo")
+
+    try:
+        sys.path = [p for p in sys.path if Path(p).resolve() != plugin_parent.resolve()]
+        if "samgeo" in sys.modules:
+            del sys.modules["samgeo"]
+
+        imported = importlib.import_module("samgeo")
+        if _is_module_from_dir(imported, plugin_pkg_dir):
+            return None
+        return imported
```

Ruff (0.16.1)

[warning] Consider moving this statement to an `else` block

(TRY300)